app/models/ability.rb

```diff
@@ -0,0 +1,15 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    if user.admin?
+      can :manage, :all
+    else
+      can :create, Post, author_id: user.id
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, user_id: user.id
+      can :read, :all
+    end
```

Comment:
- [ ] I will encourage you to make use of manage intead of create, destroy and co, manage has all in one. I encountered error trying to make comment as you can see in the image below but when I used manage, I was able to make comments and even delete comments. so just replace it with
```
 can :manage, Post, author_id: user.id
      can :manage, Comment, user_id: user.id
      can :read, :all
```
![image](https://github.com/AlselwiAisha/BlogApp/assets/86200356/f675b242-06a1-40c8-90d3-9f4111e0d1c0)

We should be good to go after this.